tests/integration/tokenomics/token_logic_modules/commutative_test.go

```diff
@@ -0,0 +1,197 @@
+package token_logic_modules
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"cosmossdk.io/log"
+	"github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/testutil/keeper"
+	"github.com/pokt-network/poktroll/testutil/sample"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	tlm "github.com/pokt-network/poktroll/x/tokenomics/token_logic_module"
+)
+
+// zerouPOKT is a coin with the uPOKT denom and zero amount, intended for use in test assertions.
+var zerouPOKT = types.NewInt64Coin(volatile.DenomuPOKT, 0)
+
+// TestTLMProcessorTestSuite asserts that the network state that results from running
+// each permutation of the default TLM processors is identical (demonstrating
+// commutativity). It does this in the following steps:
+//
+// 1. Construct a TokenomicsModuleKeepers instance for each TLM processor permutation.
+// 2. Create valid claims (which require no proofs).
+// 3. Advance the block height to the settlement height and settle the claims.
+// 5. Assert that the settlement states of all TLM order permutations match.
+func (s *tlmProcessorTestSuite) TestTLMProcessorsAreCommutative() {
+	// Generate all permutations of TLM processor ordering.
+	tokenLogicModules := tlm.NewDefaultTokenLogicModules(s.foundationBech32)
+	tlmOrderPermutations := permute(tokenLogicModules)
+
+	numTLMOrderPermutations := factorial(len(tokenLogicModules))
+	require.Equal(s.T(), numTLMOrderPermutations, len(tlmOrderPermutations))
+
+	for i, tlmPermutation := range tlmOrderPermutations {
+		var tlmIds []string
+		for _, tokenLogicModule := range tlmPermutation {
+			tlmIds = append(tlmIds, tokenLogicModule.GetId().String())
+		}
+
+		// The test description is a unique identifier for each permutation.
+		// E.g.: "permutaiton_1_of_2__TLMRelayBurnEqualsMint_TLMGlobalMint"
+		testDesc := fmt.Sprintf(
+			"permutaiton_%d_of_%d__%s",
+			i+1, numTLMOrderPermutations,
+			strings.Join(tlmIds, "_"),
+		)
+
+		s.T().Run(testDesc, func(t *testing.T) {
+			s.setupKeepers(t, keeper.WithTLMProcessors(tlmPermutation))
+
+			// Assert that no pre-existing claims are present.
+			numExistingClaims := len(s.keepers.GetAllClaims(s.ctx))
+			require.Equal(t, 0, numExistingClaims)
+
+			s.createClaims(&s.keepers, 1000)
+			settledResults, expiredResults := s.settleClaims(t)
+
+			// Set the expected state based on the effects of the first iteration;
+			// this decouples the assertions from any specific tlm effects.
```

Comment:
Thoughts on actually having a function to assert on a results to make sure we don't assert on (for example) `0=0` everywhere?

Reply:
I'm not sure I follow, can you elaborate?